### PR TITLE
feat: enhance order details with solver information

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

Screenshot 2026-02-18 at 16 09 34
 
Adds a new Solved by field to Explorer order details so users can see which solver executed an order.

  What this change does:

  - Fetches order competition status for the current order
  - Determines the executing solver from competition data
  - Renders Solved by in the Order Overview table (positioned above Hooks)
  - Enriches the display with solver metadata (logo + display name) when available
  - Falls back gracefully to text/placeholder when metadata is missing

  Correctness updates included:

  - Only attributes a solver when executedAmounts exists (no false winner fallback)
  - Re-runs solver lookup when polled order state changes (status/executed amounts/execution time), so Solved by updates without page reload

  # To Test

  1. Open an executed order in Explorer

  - [ ] Go to an order details page with a known execution (example):
    /orders/0x0e0508baaf5a7135ee7748b5516d5f051da3765e2da41a4051c5cb3035c2499dba3cb449bd2b4adddbc894d8697f5170800eadecffffffff
  - [ ] Verify a new Solved by row appears above Hooks
  - [ ] Verify solver name is shown

  2. Verify solver branding behavior

  - [ ] If solver metadata exists, verify logo + display name are rendered
  - [ ] If no logo exists, verify fallback avatar/initial is shown
  - [ ] If no solver can be resolved, verify row shows -

  3. Verify live updates for same order UID

  - [ ] Open an order that transitions from non-traded to traded while page is open
  - [ ] Verify Solved by updates automatically once execution data appears (no manual reload)

  4. Verify no false attribution

  - [ ] For orders with no executed solver data, verify Solved by stays -
  - [ ] Confirm we do not show a solver when competition entries exist but none include executedAmounts

  5. Regression checks

  - [ ] Verify existing Order Overview rows still render correctly (status, times, amount, prices, hooks, appData)
  - [ ] Verify no visual regressions in order details layout on desktop/mobile

  # Background

  Order details previously did not expose solver attribution. This PR wires order competition data into the order details view and presents the result as a dedicated Solved by field for better
  execution transparency, with safeguards to avoid stale or incorrect solver attribution.

## Summary by CodeRabbit

* **New Features**
  * Shows solver info across order views and fills: "Solved by" row, solver badges (image or initial + name), and a new "Solver" column in fills.
  * Adds hooks and UI helpers to resolve/display winning solvers (order or txHash) with loading states and fallbacks.
  * Solvers directory: rendering helpers, deeplink support (auto-expand/scroll), and new styled badge/link components.

* **Tests**
  * Expanded test suites covering solver resolution, API fallbacks, deeplink behavior, loading states, and edge cases.

### CODE REVIEW

File: apps/explorer/src/api/operator/operatorMock.ts [hunk 1]
```diff
@@ -57,3 +59,12 @@ export async function getTrades(params: GetTradesParams): Promise<RawTrade[]> {
 
   return [trade]
 }
+
+export async function getOrderCompetitionStatus(
```
Comment (shoom3301) [inline apps/explorer/src/api/operator/operatorMock.ts]: This should be a const, function doesn't use it's arguments

File: apps/explorer/src/components/orders/DetailsTable/VerboseDetails.tsx [hunk 2]
```diff
@@ -20,14 +21,33 @@ import { OrderSurplusDisplay } from '../OrderSurplusDisplay'
...
   order: Order
+  solvedBy?: OrderSolverInfo
+  isSolvedByLoading?: boolean
   isPriceInverted: boolean
   invertPrice: Command
   showFillsButton: boolean | undefined
   viewFills: Command
 }
 
+function renderSolvedBy(solvedBy?: OrderSolverInfo): ReactNode {
```
Comment (shoom3301) [inline apps/explorer/src/components/orders/DetailsTable/VerboseDetails.tsx]: render

File: apps/explorer/src/api/operator/operatorMock.ts (hunk 1 above)
Comment (fairlighteth) [reply apps/explorer/src/api/operator/operatorMock.ts]: Addressed now.

File: apps/explorer/src/components/orders/DetailsTable/VerboseDetails.tsx (hunk 2 above)
Comment (fairlighteth) [reply apps/explorer/src/components/orders/DetailsTable/VerboseDetails.tsx]: Addressed now.

**Review by elena-zh — COMMENTED** [review]
Hey @fairlighteth , thank you! Looks great!
Only one nitpick-question: is it OK that links styles are different on pages? 

Image

Image

**Comment by elena-zh** [comment]
Hey @fairlighteth , great, thank you!
The last nitpick (sorry!): might it be possible to scroll the page to top (or to a solver info) when open the page from the bottom of the page? 
For example, when I click on the link from the bottom of the page of my order details
image

I'm navigated to the bottom of the page on Solvers page: 
image

**Comment by fairlighteth** [comment]
@elena-zh added deeplinking now:


https://github.com/user-attachments/assets/ac4622b8-3037-468d-8c73-3207b6c6d28b


https://github.com/user-attachments/assets/227d6575-50eb-4506-a19e-32d2d7162525

File: apps/explorer/src/api/operator/operatorMock.ts [hunk 3]
```diff
@@ -57,3 +61,14 @@ export async function getTrades(params: GetTradesParams): Promise<RawTrade[]> {
 
   return [trade]
 }
+
+export const getOrderCompetitionStatus = async (
```
Comment (shoom3301) [inline apps/explorer/src/api/operator/operatorMock.ts:65]: The function is excessive

File: apps/explorer/src/api/operator/operatorMock.ts [hunk 4]
```diff
@@ -57,3 +61,14 @@ export async function getTrades(params: GetTradesParams): Promise<RawTrade[]> {
...
+  _params: GetOrderCompetitionStatusParams,
+): Promise<OrderCompetitionStatus | undefined> => ({
+  type: 'traded' as OrderCompetitionStatus['type'],
+  value: [{ solver: 'mock-solver', executedAmounts: { sell: '1', buy: '1' } }],
+})
+
+export const getSolverCompetitionByTxHash = async (
```
Comment (shoom3301) [inline apps/explorer/src/api/operator/operatorMock.ts:72]: The function is excessive

File: apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx [hunk 5]
```diff
@@ -1,9 +1,45 @@
...
 
 export const ALL_FILTER = 'all'
 export const ACTIVE_FILTER_ACTIVE = 'active'
 export const ACTIVE_FILTER_INACTIVE = 'inactive'
 
+function getChainIcon(chainId: number): string | undefined {
+  return CHAIN_INFO[chainId as SupportedChainId]?.logo?.light || undefined
+}
+
+function renderSolverIcon(solver: SolverInfo): React.ReactNode {
```
Comment (shoom3301) [inline apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx]: render

File: apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx [hunk 6]
```diff
@@ -87,3 +123,145 @@ export function filterSolvers(
     return visibleDeployments.length > 0 && matchesSearch(solver, query, visibleDeployments)
   })
 }
+
+function renderNetworkChips(solver: SolverInfo): React.ReactNode {
```
Comment (shoom3301) [inline apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx]: render

File: apps/explorer/src/api/operator/operatorMock.ts (hunk 3 above)
Comment (fairlighteth) [reply apps/explorer/src/api/operator/operatorMock.ts:65]: What would you propose? They are used in 'mock' mode, primarily for unit tests.

File: apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx (hunk 5 above)
Comment (fairlighteth) [reply apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx]: Damn, these keep seeping in. Fixed https://github.com/cowprotocol/cowswap/pull/7039/changes/1ed0baa919f61950a75b2160e1976a0c7200401e

File: apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx (hunk 6 above)
Comment (fairlighteth) [reply apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx]: Fixed https://github.com/cowprotocol/cowswap/pull/7039/changes/1ed0baa919f61950a75b2160e1976a0c7200401e

File: apps/explorer/src/hooks/useOrderSolver.ts [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    fetchSolversInfo(networkId).catch(() => []),
+  ])
+
+  const winnerSolverName = getWinnerSolverFromCompetition(competition)
+  if (!winnerSolverName) return undefined
+
+  return buildSolverInfo(winnerSolverName, solvers)
+}
+
+export function useTradeSolver(txHash: string | undefined | null): UseOrderSolverResult {
```
Comment (shoom3301) [inline apps/explorer/src/hooks/useOrderSolver.ts]: Nitpick: we agreed to order exported entites first:
https://nomevlabs.slack.com/archives/C04PLKRBAC8/p1771848957207429

File: apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx [hunk 8]
```diff
@@ -87,3 +123,148 @@ export function filterSolvers(
     return visibleDeployments.length > 0 && matchesSearch(solver, query, visibleDeployments)
   })
 }
+
+function networkChipsNode(solver: SolverInfo): React.ReactNode {
```
Comment (shoom3301) [inline apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx]: There are still several render functions

File: apps/explorer/src/hooks/useOrderSolver.ts (hunk 7 above)
Comment (fairlighteth) [reply apps/explorer/src/hooks/useOrderSolver.ts]: Fixed in https://github.com/cowprotocol/cowswap/pull/7039/commits/0e938340afe7aa5f948d2ad9ba01a29ca54bd5fc. I reordered useOrderSolver.ts so exported hooks are declared first and private helpers come after, in line with the agreed exports-first convention.

Please don't forget to add these rules to AGENTS.MD (in mono repo root + in cowswap-frontend) to enforce this.

File: apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx (hunk 8 above)
Comment (fairlighteth) [reply apps/explorer/src/explorer/pages/SolversDirectoryTable.helpers.tsx]: Addressed in https://github.com/cowprotocol/cowswap/pull/7039/commits/44301fc4303599dd70b29c1da7a13267321019e6 - actually turns out those helpers were not used anymore/moved but not cleaned up.

File: apps/explorer/src/hooks/useOrderSolver.ts [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+  displayName: string
+  image?: string
+}
+
+export type UseOrderSolverResult = {
+  solver: OrderSolverInfo | undefined
+  isLoading: boolean
+}
+
+export function useTradeSolver(txHash: string | undefined | null): UseOrderSolverResult {
```
Comment (shoom3301) [inline apps/explorer/src/hooks/useOrderSolver.ts]: This file is called `useOrderSolver`, so `useTradeSolver` should be defined in a separate file

Comment (fairlighteth) [reply apps/explorer/src/hooks/useOrderSolver.ts]: Addressed now in https://github.com/cowprotocol/cowswap/pull/7039/commits/c34782f482512e46fd9201a4ec0f1dec58d23f2f. I moved useTradeSolver out of useOrderSolver.ts into apps/explorer/src/hooks/useTradeSolver.ts, updated imports/tests, and extracted shared logic into orderSolverShared.ts to avoid duplication. useOrderSolver.ts now only contains the order hook.

I also added a rule in AGENTS.MD to enforce this.
